### PR TITLE
Design for kilda apps

### PR DESCRIPTION
**Pull request by timofei-durakov**

### CODE REVIEW

**Comment by timofei-durakov** [comment]
https://github.com/telstra/open-kilda/pull/2751 - PoC

File: docs/design/applications/apps.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
+# Problem
+
+OpenKilda as an SDN controller should be able to provide not only core functions but also support applications.
+This design is initial proposal for integration of one the Telescope application with controller.
+
+# Model
+
+It should be possible to add/remove application for the single flow endpoint. From DB perspective  new fields will be 
+added to `Flow` node
+2 Applications on source endpoint:
```
Comment (sergii-iakovenko) [inline docs/design/applications/apps.md]: nit: Line break needed here?

File: docs/design/applications/apps.md [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+* Remove app for the flow:  
+`DELETE /flows/:flowid/applications/:appname?switch={switch_id}&port={port_number}&vlan={vlan_id}`  
+If you do not specify an endpoint, then the application will be removed for both flow endpoints.
+
+
+## Kafka API
+
+### Messaging
+Command for the new exclusion on flow endpoint
```
Comment (sergii-iakovenko) [inline docs/design/applications/apps.md]: Please, consider describing what "exclusion" is. At this point, it's not clear.

File: docs/design/applications/apps.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+Table statistics:
+
+    {
+      "clazz": "org.openkilda.applications.info.InfoMessage",
+      "payload": {
+          "clazz": "org.openkilda.applications.info.stats.SwitchTableStatsData",
+          "switch_id": "string",
+          "table_stats_entries": [
+             {
+                "clazz": "org.openkilda.applications.info.stats.TableStatsEntry",
```
Comment (sergii-iakovenko) [inline docs/design/applications/apps.md]: nit: SwitchTableStatsEntry?

File: docs/design/applications/remove-app-for-flow-sequence.txt [hunk 4]
```diff
@@ -0,0 +1,35 @@
+@startuml
```
Comment (surabujin) [inline docs/design/applications/remove-app-for-flow-sequence.txt:1]: Use '.puml' extensin for plantum UML diagrams files please.

File: docs/design/applications/apps.md (hunk 3 above)
Comment (timofei-durakov) [reply docs/design/applications/apps.md:260]: nope, it's a tablestats, as it's done in kilda

**Pull request merged**